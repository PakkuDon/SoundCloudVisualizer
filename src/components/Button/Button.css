@import "../../styles/colors";

.root {
  background-color: $blue400;
  border-color: $blue100;
  color: $white;
  padding: 5px 10px;
  border-radius: 5px;

  &:hover {
    background-color: $blue500;
    border-color: $blue200;
  }
}
