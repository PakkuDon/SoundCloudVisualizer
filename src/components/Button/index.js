import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import styles from './styles.css'

const Button = ({ children, className, onClick, title, type }) => (
  <button
    className={classnames(styles.root, className)}
    onClick={onClick}
    title={title}
    type={type}
  >
    {children}
  </button>
)

Button.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  title: PropTypes.string,
  type: PropTypes.string,
}

Button.defaultProps = {
  className: '',
  onClick: () => {},
  title: '',
  type: 'button',
}

export default Button
