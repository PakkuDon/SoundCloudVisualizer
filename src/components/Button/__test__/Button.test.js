import React from "react"
import { render, fireEvent } from "@testing-library/react"

import Button from "../Button"

describe("Button", () => {
  it("renders children", () => {
    const children = <div>Hello world</div>
    const { container } = render(<Button>{children}</Button>)

    expect(container).toMatchSnapshot()
  })

  describe("on click", () => {
    it("calls supplied onClick handler", () => {
      const spy = jest.fn()
      const { getByText } = render(<Button onClick={spy}>Click</Button>)

      fireEvent.click(getByText("Click"))

      expect(spy).toBeCalled()
    })
  })
})
