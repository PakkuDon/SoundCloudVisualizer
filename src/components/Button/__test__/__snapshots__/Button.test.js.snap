// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders children 1`] = `
<button
  className="root"
  onClick={[Function]}
  title=""
  type="button"
>
  <div>
    Hello world
  </div>
</button>
`;
